### PR TITLE
Automated lifecycle/liveness tests

### PR DESCRIPTION
**Pull request by shaior**
This PR contains lifecycle/liveness test cases (50053, 50054, 50055, 50056, 50057, 50058)

### CODE REVIEW

**Review by theemj — APPROVED** [review]
looks good to me :+1:

File: tests/utils/pod/pod.go [hunk 1]
```diff
@@ -25,3 +26,22 @@ func RedefinePodWithLabel(pod *corev1.Pod, label map[string]string) *corev1.Pod
 
 	return pod
 }
+
+func RedefineWithLivenessProbe(pod *corev1.Pod) *corev1.Pod {
+	if len(pod.Spec.Containers) > 0 {
+		for index := range pod.Spec.Containers {
```
Comment (greyerof) [inline tests/utils/pod/pod.go:32]: Same here.

File: tests/utils/statefulset/statefulset.go [hunk 2]
```diff
@@ -39,3 +40,22 @@ func RedefineWithReplicaNumber(statefulSet *v1.StatefulSet, replicasNumber int32
 
 	return statefulSet
 }
+
+func RedefineWithLivenessProbe(statefulSet *v1.StatefulSet) *v1.StatefulSet {
+	if len(statefulSet.Spec.Template.Spec.Containers) > 0 {
+		for index := range statefulSet.Spec.Template.Spec.Containers {
```
Comment (greyerof) [inline tests/utils/statefulset/statefulset.go:46]: Same here.

File: tests/utils/deployment/deployment.go [hunk 3]
```diff
@@ -191,3 +192,22 @@ func RedefineWithNodeAffinity(deployment *v1.Deployment, key string) *v1.Deploym
 
 	return deployment
 }
+
+func RedefineWithLivenessProbe(deployment *v1.Deployment) *v1.Deployment {
+	if len(deployment.Spec.Template.Spec.Containers) > 0 {
+		for index := range deployment.Spec.Template.Spec.Containers {
```
Comment (greyerof) [inline tests/utils/deployment/deployment.go:198]: You don't need the `if len(deployment.Spec.Template.Spec.Containers) > 0`, as the `range` function won't iterate when Containers is nil or empty.

**Review by greyerof — APPROVED** [review]

**Pull request merged**